src/panels/config/backup/dialogs/dialog-download-decrypted-backup.ts

```diff
@@ -0,0 +1,228 @@
+    this._encryptionKey = "";
+    this._error = "";
+  }
+
+  protected render() {
+    if (!this._opened || !this._params) {
+      return nothing;
+    }
+
+    return html`
+      <ha-md-dialog open @closed=${this._dialogClosed} disable-cancel-action>
+        <ha-dialog-header slot="headline">
+          <ha-icon-button
+            slot="navigationIcon"
+            @click=${this.closeDialog}
+            .label=${this.hass.localize("ui.common.close")}
+            .path=${mdiClose}
+          ></ha-icon-button>
+          <span slot="title">
+            ${this.hass.localize(
+              "ui.panel.config.backup.dialogs.download.title"
+            )}
+          </span>
+        </ha-dialog-header>
+
+        <div slot="content">
+          <p>
+            ${this.hass.localize(
+              "ui.panel.config.backup.dialogs.download.description"
+            )}
+          </p>
+          <p>
+            ${this.hass.localize(
+              "ui.panel.config.backup.dialogs.download.download_backup_encrypted",
+              {
+                download_it_encrypted: html`<button
+                  class="link"
+                  @click=${this._downloadEncrypted}
+                >
+                  ${this.hass.localize(
+                    "ui.panel.config.backup.dialogs.download.download_it_encrypted"
+                  )}
+                </button>`,
+              }
+            )}
+          </p>
+
+          <ha-password-field
+            .label=${this.hass.localize(
+              "ui.panel.config.backup.dialogs.download.encryption_key"
+            )}
+            @input=${this._keyChanged}
+          ></ha-password-field>
+
+          ${this._error
+            ? html`<ha-alert alert-type="error">${this._error}</ha-alert>`
+            : nothing}
+        </div>
+        <div slot="actions">
+          <ha-button @click=${this._cancel}>
+            ${this.hass.localize("ui.dialogs.generic.cancel")}
+          </ha-button>
+
+          <ha-button @click=${this._submit}>
+            ${this.hass.localize(
+              "ui.panel.config.backup.dialogs.download.download"
+            )}
+          </ha-button>
+        </div>
+      </ha-md-dialog>
+    `;
+  }
+
+  private _cancel() {
+    this.closeDialog();
+  }
+
+  private async _submit() {
+    if (this._encryptionKey === "") {
+      return;
+    }
+    try {
+      await canDecryptBackupOnDownload(
+        this.hass,
+        this._params!.backup.backup_id,
+        this._agentId,
+        this._encryptionKey
+      );
+      downloadBackupFile(
+        this.hass,
+        this._params!.backup.backup_id,
+        this._agentId,
+        this._encryptionKey
+      );
+      this.closeDialog();
+    } catch (err: any) {
+      if (err?.code === "password_incorrect") {
+        this._error = this.hass.localize(
+          "ui.panel.config.backup.dialogs.download.incorrect_encryption_key"
+        );
+      } else if (err?.code === "decrypt_not_supported") {
```

Comment:
I think it is better if we prevent this, by doing this check before opening this dialog.

Reply:
The check is already done but I kept the error here just in case.

Reply:
It is not done in case there is no encryption key in the settings